packages/packages/commonwell-cert-runner/src/document-consumption.ts

```diff
@@ -112,15 +106,27 @@ export async function documentConsumption(commonWell: CommonWell, queryMeta: Req
   if (!patientIdForDocQuery) {
     throw new Error(`[E1c] Could not determine patient ID for document query`);
   }
-  console.log(`... patientIdForDocQuery: ${patientIdForDocQuery}`);
 
   console.log(`>>> [E1c] Querying for docs...`);
   const respDocQuery = await commonWell.queryDocument(queryMeta, patientIdForDocQuery);
-  // console.log(JSON.stringify(respDocQuery, undefined, 2));
-  if (respDocQuery.entry.length > 0) {
-    const docs = respDocQuery.entry;
-    for (const doc of docs) {
-      console.log(`... Now would download file from URL: ${doc.content.location}`);
-    }
-  }
+
+  return respDocQuery.entry;
+}
+
+export async function retrieveDocument(
+  commonWell: CommonWell,
+  queryMeta: RequestMetadata,
+  doc: Document
+): Promise<void> {
+  // E2: Document Retrieve
+  console.log(`>>> E2c: Retrieve documents using FHIR (REST)`);
+
+  const fileName = `./commonwell_${nanoid()}.file`;
+  const outputStream = fs.createWriteStream(fileName, {
+    // the default is UTF-8, avoid changing the encoding if we don't know the file we're downloading
+    encoding: null,
+  });
```

Comment:
Ideally we would pipe the stream from the download to a S3 upload, maybe when we take this to the app?

For inspiration: https://stackoverflow.com/questions/37336050/pipe-a-stream-to-s3-upload

Reply:
We'll definitely need to think about storage for these files after adding to the api. Ultimately, we'll want to process these into FHIR